### PR TITLE
Add ESM and Deployment docs

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: docs/docs/deployment.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+## Deployment Options
+
+See below for a list of deployment options for your LangChain app. If you don't see your preferred option, please get in touch and we can add it to this list.
+
+### Deploying to Fly.io
+
+[Fly.io](https://fly.io) is a platform for deploying apps to the cloud. It's a great option for deploying your app to a container environment.
+
+See [our Fly.io template](https://github.com/nfcampos/langchain-template-node-fly) for an example of how to deploy your app to Fly.io.
```
Comment (nfcampos) [inline docs/docs/deployment.md:33]: Need to transfer this to you @hwchase17

File: docs/docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  - No persistent filesystem means you can load files from disk, but not save them for later
+  - Cold start means you can't cache things in memory and expect them to be cached between requests
+  - Function timeouts mean you can't support long-running requests, such as WebSockets
+
+Some other considerations include:
+
+- Do you deploy your backend and frontend together, or separately?
+- Do you deploy your backend co-located with your database, or separately?
+
+## Deployment Options
```
Comment (nfcampos) [inline docs/docs/deployment.md:25]: I want to add more, but we can start with one?

File: docs/docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - You can support long-running requests, such as WebSockets
+- Deploying to a serverless environment
+  - No persistent filesystem means you can load files from disk, but not save them for later
+  - Cold start means you can't cache things in memory and expect them to be cached between requests
+  - Function timeouts mean you can't support long-running requests, such as WebSockets
+
+Some other considerations include:
+
+- Do you deploy your backend and frontend together, or separately?
+- Do you deploy your backend co-located with your database, or separately?
```
Comment (nfcampos) [inline docs/docs/deployment.md:23]: I thought about adding a line here about "If you want more support on getting your langchain app in production get in touch and we're happy to help" thoughts @hwchase17 ?

File: docs/docs/deployment.md (hunk 2 above)
Comment (hwchase17) [reply docs/docs/deployment.md:25]: perfect

File: docs/docs/deployment.md (hunk 1 above)
Comment (hwchase17) [reply docs/docs/deployment.md:33]: accepted

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**